nml/actions/action0properties.py

```diff
@@ -1058,76 +686,583 @@ def label_list(value, prop_num, description):
     return [LabelListProp(prop_num, value.values)]
 
 common_tracktype_props = {
-    'label'                    : {'size': 4, 'num': 0x08, 'string_literal': 4}, # is allocated during reservation stage, setting label first is thus not needed
-    'toolbar_caption'          : {'size': 2, 'num': 0x09, 'string': 0xDC},
-    'menu_text'                : {'size': 2, 'num': 0x0A, 'string': 0xDC},
-    'build_window_caption'     : {'size': 2, 'num': 0x0B, 'string': 0xDC},
-    'autoreplace_text'         : {'size': 2, 'num': 0x0C, 'string': 0xDC},
-    'new_engine_text'          : {'size': 2, 'num': 0x0D, 'string': 0xDC},
-    'construction_cost'        : {'size': 2, 'num': 0x13},
-    'map_colour'               : {'size': 1, 'num': 0x16},
-    'introduction_date'        : {'size': 4, 'num': 0x17},
-    'sort_order'               : {'size': 1, 'num': 0x1A},
-    'name'                     : {'size': 2, 'num': 0x1B, 'string': 0xDC},
-    'maintenance_cost'         : {'size': 2, 'num': 0x1C},
+    'label':                {'size': 4, 'num': 0x08, 'string_literal': 4}, # is allocated during reservation stage, setting label first is thus not needed
+    'toolbar_caption':      {'size': 2, 'num': 0x09, 'string': 0xDC},
+    'menu_text':            {'size': 2, 'num': 0x0A, 'string': 0xDC},
+    'build_window_caption': {'size': 2, 'num': 0x0B, 'string': 0xDC},
+    'autoreplace_text':     {'size': 2, 'num': 0x0C, 'string': 0xDC},
+    'new_engine_text':      {'size': 2, 'num': 0x0D, 'string': 0xDC},
+    'construction_cost':    {'size': 2, 'num': 0x13},
+    'map_colour':           {'size': 1, 'num': 0x16},
+    'introduction_date':    {'size': 4, 'num': 0x17},
+    'sort_order':           {'size': 1, 'num': 0x1A},
+    'name':                 {'size': 2, 'num': 0x1B, 'string': 0xDC},
+    'maintenance_cost':     {'size': 2, 'num': 0x1C},
 }
```

Comment:
I think it's a mistake to move this away from the road/rail/tramtype blocks it's used by -- it's basically part of those, and anyone reading the code will want to see both at the same time.

[for context, each of those blocks contained a copy-pasted version of this until I factored it out recently-ish]

It's also now above the comment that explains what it means.

Reply:
Oh, but `properties` now being a huge list literal means you can't do that easily.

To be honest, I'm not convinced this commit is an improvement -- a list with arbitrary 'None' holes isn't clearer than the dict, especially when you have to put the indices in comments anyway because they're semantically meaningful.

Ideally we should use the feature names instead of the hex values, I think there's a patch for that in one of @planetmaker 's closed PRs.

Reply:
You could be right. The intention was to remove the number of `#fmt: off/on` comments. As TB has said, perhaps the best solution longterm is to move all of these large tables of data into a separate folder that can be properly excluded from checks, as data is not code (like was done for eints)